### DISCUSSION TITLE
Feature Request: Inactive Flag on Rollup

### DISCUSSION THREAD

**Discussion by dschach** [jamessimone/apex-rollup#482] [Ideas]
For backwards compatibility, it isn't feasible to have an Active flag, but it would be nice to have an inactive flag so we can create rollups and have them ignored while they're works in progress. 

It will help for deploying from sandboxes to production in enterprise environments where everything has to be turned on at once, for example.

**Comment by jamessimone** [comment]
There are two levels of "active" at the moment, with an optional third on Rollup already (the "should ignore custom setting..." flag):

- Should Abort Run on the rollup control record the rollup(s) are tied to
- the Rollup Settings hierarchical custom settings record for the org/profile/user

If you were looking to do a "big bang" style move from one environment to another, I'd recommend tying said rollups to a rollup control record where the Should Abort Run flag is set to true. You can then either pluck individual rollups to an "Active" rollup control record as desired, or simply update the flag on the existing control record when you'd like to enable all of the rollups.

**Reply by baobao917** [reply]
I was interested in this feature request as well.  I couldn't find the "optional third on Rollup already (the "should ignore custom setting..." flag).  Below is a screenshot of one of my rollups and I don't see this "should ignore custom setting..."

![image](https://github.com/jamessimone/apex-rollup/assets/104730171/4356c6f6-71ed-4918-b4f4-320e6089e7ad)

**Reply by jamessimone** [reply]
You may need to (temporarily) add it to the Rollup__mdt layout, @baobao917. That flag hasn't been exposed to end users because it's primarily designed for namespace package users bundling Apex Rollup into their own packages.

I'd be curious to hear what you're trying to accomplish here that can't already be done by tying a rollup you're looking to not have active to a separate rollup control or through the custom setting?

**Reply by baobao917** [reply]
@jamessimone I was just trying to understand all the different options.  I think tying to a separate rollup control or through the custom setting will work.  It just wasn't as intuitive compared to DLRS where each roll-up has an "Active" checkbox.  Thanks!
